server/routes/api/v1/allergies/index.js

```diff
@@ -0,0 +1,41 @@
+import { Role } from '../../../../models/user.js';
+
+export default async function (fastify) {
+  fastify.get(
+    '',
+    {
+      schema: {
+        querystring: {
+          allergy: { type: 'string' },
+        },
+      },
+      onRequest: fastify.requireUser([Role.ADMIN, Role.STAFF, Role.VOLUNTEER]),
+    },
+    async (request, reply) => {
+      const allergy = request.query.allergy.trim();
+
+      // prevent empty string from returning all allergies
+      if (!allergy.length) {
+        return reply.send({ message: 'No query provided' });
+      }
+
+      const NO_ALLERGIES = ['none', 'no allergies', 'no known allergies', 'n/a'];
+
+      if (NO_ALLERGIES.includes(allergy.toLowerCase())) {
+        return reply.send({ message: 'No known allergies' });
+      }
```

Comment:
Similar concern as the situation with no query parameter being provided

Reply:
I'm wondering if we need to handle this input... instead, maybe we have a boolean in the Patient record for "no allergies" (and the others) that the user can select with a checkbox, or the user just leaves the section blank